test/SparqlAntlrParserTest.cpp

```diff
@@ -855,3 +855,33 @@ TEST(SparqlParser, RDFLiteral) {
       "\"1950-01-01T00:00:00\"^^<http://www.w3.org/2001/XMLSchema#dateTime>");
   expectRDFLiteralFails(R"(?a ?b "The \"Moon\""@en .)");
 }
+
+TEST(SparqlParser, SelectQuery) {
+  auto expectSelectQuery = ExpectCompleteParse<&Parser::selectQuery>{
+      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}}};
+  auto expectSelectQueryFails = ExpectParseFails<&Parser::selectQuery>{};
+  expectSelectQuery(
+      "SELECT * WHERE { ?a <bar> ?foo }",
+      m::ParsedQuery("SELECT * WHERE { ?a <bar> ?foo }", m::AsteriskSelect(),
+                     m::GraphPattern(m::Triples({{"?a", "<bar>", "?foo"}}))));
+  expectSelectQuery(
+      "SELECT ?foo WHERE { } GROUP BY ?foo",
+      m::ParsedQuery("SELECT ?foo WHERE { } GROUP BY ?foo",
+                     m::VariablesSelect({"?foo"}), m::GraphPattern(), {}, {},
+                     {Variable{"?foo"}}, {}));
+  expectSelectQuery(
+      "SELECT (COUNT(?foo) as ?baz) WHERE { } GROUP BY ?bar",
+      m::ParsedQuery("SELECT (COUNT(?foo) as ?baz) WHERE { } GROUP BY ?bar",
+                     m::Select({std::pair{"COUNT(?foo)", "?baz"}}),
+                     m::GraphPattern(), {}, {}, {Variable{"?bar"}}));
+  // `SELECT *` is not allowed while grouping.
+  expectSelectQueryFails("SELECT * WHERE { } GROUP BY ?foo");
+  // `?foo` is selected twice. Once as variable and once as the result of an
+  // alias. This is not allowed.
+  expectSelectQueryFails("SELECT ?foo (?bar as ?foo) WHERE { }");
+  // When grouping selected variables must either be grouped by or aggregated.
+  // `?foo` is neither.
+  expectSelectQueryFails("SELECT (?bar as ?foo) WHERE { } GROUP BY ?baz");
+  // Datasets are not supported.
+  expectSelectQueryFails("SELECT * FROM  WHERE <foo> { }");
```

Comment:
Add checks for
* The new failure cases I have commented on above

* The (valid) selection of a simple grouped variable. I think you currently only have a test for aliases.

Reply:
And in general I think there are some cases missing, because some arguments to your `ParsedQuery` matchers (OrderBy, Having) etc. are currently never used.

Reply:
The selection of a grouped variable is here: https://github.com/ad-freiburg/qlever/blob/e40ab7db49bda0496f01a23bc189a2e2fff01c16/test/SparqlAntlrParserTest.cpp#L875-L879